src/analysis/normed_space/lp_space.lean

```diff
@@ -0,0 +1,516 @@
+
+lemma is_lub_norm [nonempty α] (f : lp E ∞) : is_lub (set.range (λ i, ∥f i∥)) ∥f∥ :=
+begin
+  rw lp.norm_eq_csupr,
+  exact is_lub_csupr (lp.mem_ℓp f)
+end
+
+lemma norm_eq_tsum_rpow (hp : 0 < p.to_real) (f : lp E p) :
+  ∥f∥ = (∑' i, ∥f i∥ ^ p.to_real) ^ (1/p.to_real) :=
+begin
+  dsimp [norm],
+  rw ennreal.to_real_pos_iff at hp,
+  rw [dif_neg hp.1.ne', if_neg hp.2.ne],
+end
+
+lemma norm_rpow_eq_tsum (hp : 0 < p.to_real) (f : lp E p) :
+  ∥f∥ ^ p.to_real = ∑' i, ∥f i∥ ^ p.to_real :=
+begin
+  rw [norm_eq_tsum_rpow hp, ← real.rpow_mul],
+  { field_simp [hp.ne'] },
+  apply tsum_nonneg,
+  intros i,
+  calc (0:ℝ) = 0 ^ p.to_real : by rw real.zero_rpow hp.ne'
+  ... ≤ _ : real.rpow_le_rpow rfl.le (norm_nonneg (f i)) hp.le
+end
+
+lemma has_sum_norm (hp : 0 < p.to_real) (f : lp E p) :
+  has_sum (λ i, ∥f i∥ ^ p.to_real) (∥f∥ ^ p.to_real) :=
+begin
+  rw norm_rpow_eq_tsum hp,
+  exact ((lp.mem_ℓp f).summable hp).has_sum
+end
+
+lemma norm_nonneg' (f : lp E p) : 0 ≤ ∥f∥ :=
+begin
+  rcases p.trichotomy with rfl | rfl | hp,
+  { simp [lp.norm_eq_card_dsupport f] },
+  { cases is_empty_or_nonempty α with _i _i; resetI,
+    { rw lp.norm_eq_csupr,
+      simp [real.csupr_empty] },
+    inhabit α,
+    exact (norm_nonneg (f (default α))).trans ((lp.is_lub_norm f).1 ⟨default α, rfl⟩) },
+  { rw lp.norm_eq_tsum_rpow hp f,
+    refine real.rpow_nonneg_of_nonneg (tsum_nonneg _) _,
+    exact λ i, real.rpow_nonneg_of_nonneg (norm_nonneg _) _ },
+end
+
+@[simp] lemma norm_zero : ∥(0 : lp E p)∥ = 0 :=
+begin
+  rcases p.trichotomy with rfl | rfl | hp,
+  { simp [lp.norm_eq_card_dsupport] },
+  { simp [lp.norm_eq_csupr] },
+  { rw lp.norm_eq_tsum_rpow hp,
+    have hp' : 1 / p.to_real ≠ 0 := one_div_ne_zero hp.ne',
+    simpa [real.zero_rpow hp.ne'] using real.zero_rpow hp' }
+end
+
+lemma norm_eq_zero_iff ⦃f : lp E p⦄ : ∥f∥ = 0 ↔ f = 0 :=
+begin
+  classical,
+  refine ⟨λ h, _, by { rintros rfl, exact norm_zero }⟩,
+  rcases p.trichotomy with rfl | rfl | hp,
+  { ext i,
+    have : {i : α | ¬f i = 0} = ∅ := by simpa [lp.norm_eq_card_dsupport f] using h,
+    have : (¬ (f i = 0)) = false := congr_fun this i,
+    tauto },
+  { cases is_empty_or_nonempty α with _i _i; resetI,
+    { simp },
+    have H : is_lub (set.range (λ i, ∥f i∥)) 0,
+    { simpa [h] using lp.is_lub_norm f },
+    ext i,
+    have : ∥f i∥ = 0 := le_antisymm (H.1 ⟨i, rfl⟩) (norm_nonneg _),
+    simpa using this },
+  { have hf : has_sum (λ (i : α), ∥f i∥ ^ p.to_real) 0,
+    { have := lp.has_sum_norm hp f ,
+      rw h at this,
+      simpa [real.zero_rpow hp.ne'] using this }, -- why can't the `simp` and `rw` be combined?
+    have : ∀ i, 0 ≤ ∥f i∥ ^ p.to_real := λ i, real.rpow_nonneg_of_nonneg (norm_nonneg _) _,
+    rw has_sum_zero_iff_of_nonneg this at hf,
+    ext i,
+    have : f i = 0 ∧ p.to_real ≠ 0,
+    { simpa [real.rpow_eq_zero_iff_of_nonneg (norm_nonneg (f i))] using congr_fun hf i },
+    exact this.1 },
+end
+
+lemma eq_zero_iff_ae_eq_zero {f : lp E p} : f = 0 ↔ ⇑f = 0 :=
```

Comment:
Is this a copy-paste that fell between the cracks?

How about:
```suggestion
lemma eq_zero_iff_coe_fn_eq_zero {f : lp E p} : f = 0 ↔ ⇑f = 0 :=
```

Reply:
Ha, yes :)